pkg/controller/webhooks/deletionprotection.go

```diff
@@ -50,26 +51,29 @@ func checkManagedObject(ctx context.Context, obj client.Object, c client.Client,
 
 	ownerKind, ok := obj.GetLabels()[runtime.OwnerKindAnnotation]
 	if !ok || ownerKind == "" {
-		l.Info(runtime.OwnerKindAnnotation + " label not set, skipping evaluation")
-		return compositeInfo{Exists: false}, nil
+		l.Info(runtime.OwnerKindAnnotation + " label not set, blocking deletion")
+		return compositeInfo{Exists: isDeletionProtected(obj), Reason: fmt.Sprintf("%s label not set, blocking deletion", runtime.OwnerKindAnnotation)}, nil
```

Comment:
Didn't we say we block here the deletion?

Reply:
It gets blocked.

`isDeletionProtected(obj)` checks if the override label is set or not. This way we can get things unstuck if for any reason the labels get messed up completely.